### ISSUE TITLE
how to apply a global Interceptor

### ISSUE THREAD

**Issue by xujif** [nestjs/nest#159]
I want to trasfrom all response ,  like  classToPlain to expose or exclude some field instead of JSON.stringify.

**Comment by cojack** [comment]
@xujif did you check Pipes? http://docs.nestjs.com/pipes

**Comment by xujif** [comment]
@cojack pips seems can only  access request (query,body) can not access response.
Interceptor can do this ,but there's no "useGlobalInterceptors”

**Comment by adrien2p** [comment]
You can use a function to do this and use it in the express instance as instance.use(interceptor)

**Comment by xujif** [comment]
I added the code wich bootstrap and worked.
```typescript
 app.use((req, res, next) => {
        var oldJson = res.json;
        res.json = function (data) {
            if (data && typeof data === 'object') {
                data = classToPlain(data)
            }
            arguments[0] = data;
            oldJson.apply(res, arguments);
        }
        next()
    })
```
but it's ugly, can you add a "useGlobalInterceptors" method?

**Comment by xujif** [comment]
like 
```typescript
import { Interceptor, NestInterceptor, ExecutionContext, UseInterceptors } from '@nestjs/common';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/operator/map';
import { classToPlain } from 'class-transformer'

@Interceptor()
export class TransformInterceptor implements NestInterceptor {
    intercept (dataOrRequest, context: ExecutionContext, stream$: Observable<any>): Observable<any> {
        return stream$.map((data) => (classToPlain(data)));
    }
}

//  app.useGlobalInterceptors(new TransformInterceptor ());
```

**Comment by cojack** [comment]
@xujif but why do you even wanna to access response when Classes can and should cover `toJSON` method and there return plain object that can be easily converted to JSON?

**Comment by xujif** [comment]
classToPlain supports some useful feture: like  expose or exclude , getters, version...

**Comment by xujif** [comment]
see https://github.com/pleerock/class-transformer/blob/master/README.md

**Comment by cojack** [comment]
@xujif you don't understand me, try something like this:
```typescript
import {Exclude, classToPlain} from "class-transformer";

export default class User {

    id: number;

    email: string;

    @Exclude()
    password: string;

    toJSON() {
        return classToPlain(this);
    }
}
```

And then, you have a powerful control of all of your entities in your app.

**Comment by xujif** [comment]
It‘s a good solution. But some times , Global Interceptor can do more. I also suggest to add it. 
If you don't think so , close the issue.

**Comment by cojack** [comment]
@xujif but cache is specific interceptor, and you can not set cache as global one. How do you wanna cache create entities? It's nonsens. And even, CACHE should be implemented on REQUEST level, not at RESPONSE level, and with that you can do with interceptor on Controller as you can see it in examples 01-cats-app in cats.controller.ts.

But I agree that useGlobalInterceptor will be cool

**Comment by xujif** [comment]
I  just found the code from http://docs.nestjs.com/interceptors ,but it can not apply to global
```typescript
import { Interceptor, NestInterceptor, ExecutionContext } from '@nestjs/common';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/of';

@Interceptor()
export class CacheInterceptor implements NestInterceptor {
  intercept(dataOrRequest, context: ExecutionContext, stream$: Observable<any>): Observable<any> {
    const isCached = true;
    if (isCached) {
      return Observable.of([]);
    }
    return stream$;
  } 
}
```

**Comment by kamilmysliwiec** [comment]
Hi @xujif,
Good idea! It's available since `4.0.1`. Please, update your packages :slightly_smiling_face:
